From 89cd60c088fe98ee0832b43610c0374337967f5a Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 11 Mar 2021 11:33:45 +0200
Subject: [PATCH 1321/1921] arch/arm64: dts: disable SMMU on Armada platforms

Currently SMMU causes problems with DPDK hugebuffer usage.
Disable it in common builds for now.

Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Change-Id: I554d9f57146b78ee3276c393a94f761217026b9d
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/47664
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040.dtsi | 2 +-
 arch/arm64/boot/dts/marvell/armada-8040.dtsi | 2 +-
 arch/arm64/boot/dts/marvell/cn9130.dtsi      | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index 7a3198cd7a07..0cbba6e891ff 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -16,7 +16,7 @@
 };
 
 &smmu {
-	status = "okay";
+	status = "disabled";
 };
 
 &cp0_pcie0 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040.dtsi b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
index 79e8ce59baa8..101be8fa68bd 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
@@ -16,7 +16,7 @@
 };
 
 &smmu {
-	status = "okay";
+	status = "disabled";
 };
 
 &cp0_pcie0 {
diff --git a/arch/arm64/boot/dts/marvell/cn9130.dtsi b/arch/arm64/boot/dts/marvell/cn9130.dtsi
index 0835e3af2de3..340b18015ef4 100644
--- a/arch/arm64/boot/dts/marvell/cn9130.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130.dtsi
@@ -39,7 +39,7 @@
 #undef CP11X_PCIE2_BASE
 
 &smmu {
-	status = "okay";
+	status = "disabled";
 };
 
 &cp0_sata0 {
-- 
2.31.1

